File: 198. House Robber.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        dp=list()
        if len(nums)==1:
            return nums[0]
        if len(nums)==0:
            return 0
        dp.append(nums[0])
        dp.append(max(nums[0],nums[1]))
        output=0
        for i in range(2,len(nums)):
            dp.append(max(dp[i-1],nums[i]+dp[i-2]))
        print(dp)
        return dp[-1]

File: 198_House_Robber.cpp
class Solution {
public:
    int rob(vector<int>& nums) {
        if(nums.empty())
            return 0;
        if(nums.size()==1)
            return nums[0];
        if(nums.size()==2)
            return max(nums[0],nums[1]);
        vector<int> totalvalue;
        totalvalue.push_back(nums[0]);
        totalvalue.push_back(max(nums[0],nums[1]));
        for(int i=2;i<nums.size();i++)
        {
            totalvalue.push_back(max(nums[i]+totalvalue[i-2],totalvalue[i-1]));
        }
        return totalvalue.back();
        
    }
};